src/providers/ogr/qgsogrfeatureiterator.cpp

```diff
@@ -106,8 +102,16 @@ QgsOgrFeatureIterator::QgsOgrFeatureIterator( QgsOgrFeatureSource *source, bool
   if ( !mRequest.filterRect().isNull() )
   {
     const QgsRectangle &rect = mRequest.filterRect();
-
+#if 0
```

Comment:
This is still a problem which must be looked into at some point.

> // rounding errors with 2nd geometry of provider/gdal_220.autotest.ogr_multiplegeomfields.gml

In your sample, the second POINT had a value of Point(2,3)
* it was not displayed in QGIS because the returned extent was not exactly 2.0,3.0
   * but slightly less or more - I assume a typical **double** problem.